batch-job.go

```diff
@@ -191,6 +212,57 @@ const BatchJobExpireTemplate = `expire:
     delay: 500ms # least amount of delay between each retry
 `
 
+const BatchJobCatalogTemplate = `catalog:
+  apiVersion: v1
+  id: mycatjob # unique identifier for the catalog job
+
+  # The source bucket to list and catalog.
+  bucket: mysourcebucket
+
+  # destination info for catalog output objects.
+  destination:
+    bucket: mybucket
+    prefix: myprefix # optional prefix ('/' will be appended if not present)
+    format: ndjson # csv or ndjson (newline delimited json)
+
+  # scheduling info for the catalog job: valid values are:
+  #   "once|daily|weekly|monthly|yearly"
+  #
+  # once: starts immediately and runs only once
+  #
+  # All the following schedules do not start the job immediately on submission.
+  # daily: runs every day at roughly the same time
+  # weekly: runs every sunday at roughly the same time
+  # monthly: runs on first sunday of every month.
+  # yearly: runs on first sunday of January every year.
+	#
+  # FIXME: only once supported for initial implementation
+  schedule: "once"
+
+  # Mode provides a tradeoff between speed and consistency:
+  #   "fast" -> use fewer resources and complete faster, but some output data
+  #     may be stale. This is better for tasks where approximate results are
+  #     acceptable. For example: storage usage
+  #   "strict" -> use more resources and complete slower, but avoids returning
+  #     stale data.
+  # In either case, objects modified (created, replaced, removed) during the
+  # catalog job run may not be included in the output.
+  mode: fast
+
+  # "versions" specifies if only current or all versions of each object should
+  # be processed.
+  versions: current # Other valid value is "all"
+
+  # The list of optional output fields to include. Please refer to documentation
+  # for the full list of default and optional fields available.
+  includeFields:
+    - ETag
+    - IsMultipartUpload
```

Comment:
Checksum? Tags? Those would be really nice to have (feel free to close, since this is pretty irrelevant here)

Reply:
Yeah, the full list will be added in docs. The YAML file template already looks quite large and more will be added.